andela/ah-kgl-avengers-backend
# #165020206-update user profile

## What does this PR do?
- creates the endpoint for the user to update his Profile.
- creates the endpoint for the user to view other's profiles but he/she can't edit them.
- implement the functionality to store the user profile image on the cloud.

## Description of Task to be completed?
- Implement the User update the Profile
- Implement the User view other's profiles.

## How should this be manually tested?
Set up the project following the README.md.
Using an API client like Postman or Insomnia:
- You first need to signup by Making a POST request to localhost:3000/api/v1/auth/signup with a JSON body that includes username,email and password fields with corresponding values:
{ "username": "username", email : your.email@address.com, "password": YourPassword }
You should get a 201 created response if all data are valid
- Second You will require to login and get the token to be authenticated by Making a POST request to localhost:3000/api/v1/auth/login with a JSON body that includes email and password fields with corresponding values:
{ "email" : your.email@address.com, "password": YourPassword }
You should get a 200 OK response if all data are valid.
- To test the profile updating endpoint Make a PUT request to `localhost:3000/api/v1/users/profile/:username/update` with a JSON body that includes  `username`,`bio` and `image` fields with corresponding values and add token in headers: 
{ "username": "username", `bio` : `your bio`,  `image`: `image text as form of file` }
 You should get a `200 created` response if all data are valid.

- To test the View other's profiles endpoint Make a GET request to `localhost:3000/api/v1/users/profile/:username`, You should get a `200 OK` response and all data of a particular user and add token in headers.
## Any background context you want to provide?
You will need to set up a .env file containing the Facebook and Google OAuth Client IDs, tokens, Cloud name, Cloud API KEY, and CLOUD API SECRET.
## What are the relevant pivotal tracker stories?
[165020206](https://www.pivotaltracker.com/n/projects/2321840)
